beluga/include/beluga/sensor/data/landmark_map.hpp

```diff
@@ -46,9 +47,23 @@ class LandmarkMap {
   /// @brief Constructor.
   /// @param boundaries Limits of the map.
   /// @param landmarks List of landmarks that can be expected to be detected.
-  LandmarkMap(const LandmarkMapBoundaries& boundaries, landmarks_set_position_data landmarks)
+  explicit LandmarkMap(const LandmarkMapBoundaries& boundaries, landmarks_set_position_data landmarks)
```

Comment:
The explicit boundaries are kind of a necessary evil. The only reason this exists is to initialize the `beluga::MultivariateUniformDistribution uniform_state_distribution_`, and the thing with Landmark maps, which does not happen in the case of OccupancyMaps, is that the support of the map may be much smaller than the traversable region.

- a field with scattered apriltags (the robot can still navigate on the outside of the region delimited by the furthest landmarks.
- all landmarks painted on the surface of a single flat wall (the landmarks delimit a zero area region, with the robots on the outside).
- four apriltags on top of a single pole, looking different ways (again zero region, and all "random" states will be a single point).

This needs not cause a failure under normal behavior, but no random states will be generated outside of the area "enclosed" be landmarks.

There's definitely a subset of cases where the apriltags region is a superset of the traversable region (the demo in beluga-demos, for instance), but at the very best a word of warning should be given in the comments for users that decide to use this constructor.

Reply:
Agreed. I'll rephrase the above for users to know what equating boundaries to map support means in practice.

Reply:
See https://github.com/Ekumen-OS/beluga/pull/419/commits/76f6c1be9e6074ebdf3bcf759d8a5e91434c7e8a.